ykpack/src/types.rs

```diff
@@ -81,6 +81,65 @@ impl Display for Local {
     }
 }
 
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Hash)]
+pub struct Place {
+    pub base: PlaceBase,
+    pub projections: Vec<PlaceProjection>,
+}
+
+impl Display for Place {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.projections.is_empty() {
+            write!(f, "{}", self.base)?;
+        } else {
+            write!(f, "({})", self.base)?;
+            for p in &self.projections {
+                write!(f, "{}", p)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+impl From<Local> for Place {
+    fn from(l: Local) -> Self {
+        Self {
+            base: PlaceBase::Local(l),
+            projections: Vec::new(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Hash)]
+pub enum PlaceBase {
+    Local(Local),
+    Static, // FIXME not implemented
```

Comment:
Should we fix this?

Reply:
I think we should implement that in a future PR.

There are lots of language features that we don't yet support which we can incrementally implement as needed.

It's like your gameboy emulator. Implementing all opcodes at once would kill you.

Reply:
Point taken!